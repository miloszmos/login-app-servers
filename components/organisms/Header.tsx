import React from 'react';
import { useLogout } from '../../hooks/useLogout';

const Header = () => {
  const { logout, loading } = useLogout();
  return (
    <div className="h-20 bg-neutral-800 shadow-md md:shadow-xl w-full flex items-center">
      <div className="container mx-auto flex justify-between px-6">
        <p className="text-3xl text-slate-200">_Servers</p>
        <button
          aria-label="logout"
          disabled={loading}
          aria-disabled={loading}
          type="button"
          onClick={logout}
          className="text-slate-200 font-semibold underline decoration-transparent underline-offset-1 transition ease-in-out hover:decoration-white duration-300"
        >
          {loading ? 'See You!' : <>Logout &rarr;</>}
        </button>
      </div>
    </div>
  );
};

export default Header;
