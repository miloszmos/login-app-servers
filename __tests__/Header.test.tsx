import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Header from '../components/organisms/Header';

describe('<Header />', () => {
  it('renders and matches the snapshot', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
  test('renders a header title', () => {
    const { getByText } = render(<Header />);
    expect(getByText('_Servers')).toBeInTheDocument();
  });
  test('shows a logout button', () => {
    const { getByLabelText } = render(<Header />);
    expect(getByLabelText('logout')).toBeInTheDocument();
    expect(getByLabelText('logout')).toHaveTextContent('Logout â†’');
  });
});
