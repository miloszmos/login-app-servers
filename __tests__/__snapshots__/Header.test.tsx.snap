// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders and matches the snapshot 1`] = `
<div>
  <div
    class="h-20 bg-neutral-800 shadow-md md:shadow-xl w-full flex items-center"
  >
    <div
      class="container mx-auto flex justify-between px-6"
    >
      <p
        class="text-3xl text-slate-200"
      >
        _Servers
      </p>
      <button
        aria-disabled="false"
        aria-label="logout"
        class="text-slate-200 font-semibold underline decoration-transparent underline-offset-1 transition ease-in-out hover:decoration-white duration-300"
        type="button"
      >
        Logout â†’
      </button>
    </div>
  </div>
</div>
`;
